net/lp_rpc.proto

```diff
@@ -38,6 +88,9 @@ message TranscoderInfo {
 
   // Transcoded streamId list to update the master manifest on the broadcaster.
   map<string, string> streamIds = 16;
+
+  // Orchestrator returns info about own input object storage, if it wants it to be used.
+  OSInfo preferredIOS = 32;
```

Comment:
Makes sense. What do you think about making the `OSInfo` struct `repeated` following https://github.com/livepeer/go-livepeer/issues/582 ?

Reply:
It was repeated at first, I've remove `repeated` for the sake of simplicity and because we wanted to move negotiation to the next step.

Reply:
One reason we might want to make it `repeated` now is because we can't really change fields to `repeated` in a backwards compatible way later on. So when we need to support negotiation and/or improve support for concurrent OS, we'd have to make a new field to accommodate that, and this field becomes unnecessary (but we'd still have to carry it around for some time). So if we can future-proof things a little bit by marking it `repeated`, then that's a win.

Also, an empty array is a useful way to indicate "no storage preference" so we don't have to encode that value explicitly.